src/quml-to-odk/quml-to-odk.service.ts

```diff
@@ -0,0 +1,404 @@
+        const option = question.editorState.options[i];
+        if (option.answer) {
+          correctOption = optionsMap[i];
+        }
+        choicesSheetArray.push([
+          itemName,
+          optionsMap[i],
+          QumlToOdkService.cleanHtml(option.value.body),
+        ]); // populate choices sheet
+      }
+      const optionItemName = itemName + '_ans';
+      totalMarksCellArray.push('${' + optionItemName + '}');
+      optionRowsArray.push([
+        'calculate',
+        optionItemName,
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        'if(${' + itemName + '} = ‘' + correctOption + "‘, '1', '0')",
+        '',
+      ]);
+
+      index++; // at last increment the index
+    }); // forEach() ends
+
+    surveySheetArray.push([
+      'end group',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+    ]);
+
+    surveySheetArray = surveySheetArray.concat(optionRowsArray);
+
+    // push hidden `total_marks` row to $optionRows
+    surveySheetArray.push([
+      'calculate',
+      'total_marks',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      '',
+      totalMarksCellArray.join(' + '),
+      '',
+    ]);
+
+    // push the remaining rows
+    surveySheetArray = surveySheetArray.concat([
+      [
+        'hidden',
+        'ques_number',
+        questions.result.questions.length,
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+      ],
+      [
+        'hidden',
+        'total_ques',
+        'कुल प्रश्न = ' + questions.result.questions.length,
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+      ],
+      [
+        'hidden',
+        'total_correct',
+        'कुल प्रश्न जिसका सही उत्तर दिया ${total_marks}',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+        '',
+      ],
+      ['start', 'start', '', '', '', '', '', '', '', '', '', ''],
+      ['end', 'end', '', '', '', '', '', '', '', '', '', ''],
+      ['today', 'today', '', '', '', '', '', '', '', '', '', ''],
+    ]);
+
+    settingsSheetArray.push([name, 'g2m_w1', 'no']);
+
+    return [surveySheetArray, choicesSheetArray, [], settingsSheetArray];
+  }
+
+  /**
+   * Returns the absolute path of the .xlsx file created
+   * @param surveySheetArray
+   * @param choicesSheetArray
+   * @param mediaSheetArray
+   * @param settingsSheetArray
+   *
+   * @return string file (absolute path of the generated XSLX form)
+   */
+  private createXslx(
+    surveySheetArray: Array<any>,
+    choicesSheetArray: Array<any>,
+    mediaSheetArray: Array<any>,
+    settingsSheetArray: Array<any>,
+  ): string {
+    const filename = +new Date() + '.xlsx';
```

Comment:
create a UUID based filename.

Reply:
It's done. Reference commit: https://github.com/Samagra-Development/quml2xform/pull/1/commits/41c603d1e89e549c052e409f47a60e9ffa6b6f4e